### DISCUSSION TITLE
Too painful the app dir... switching to pages

### DISCUSSION THREAD

**Discussion by vitalijalbu** [vercel/next.js#65958] [App Router]
I know **SSR** can be a "faster" solution but maan, I spent a week and still cannot create a simple datatable with dummy JSON (using  [rc-table](https://www.npmjs.com/package/rc-table))... page is server side, buttons to edit must be client, datatable component also must be client 😅... if you have a modal to edit and save, it's a suicide...
if you want to create **re-usable components** app dir is not the solution...

no comment... maybe I'm wrong, but show me a real world example with complete SSR 👀

**Comment by SouthLink** [comment]
Check out this official example : https://github.com/vercel/commerce
